src/node_code_integrity.cc

```diff
@@ -0,0 +1,279 @@
+#include "node_code_integrity.h"
+#include "v8.h"
+#include "node.h"
+#include "env-inl.h"
+#include "node_external_reference.h"
+
+namespace node {
+
+using v8::Boolean;
+using v8::Context;
+using v8::FunctionCallbackInfo;
+using v8::Local;
+using v8::Object;
+using v8::Value;
+
+namespace codeintegrity {
+
+#ifdef _WIN32
+static bool isWldpInitialized = false;
+static pfnWldpCanExecuteFile WldpCanExecuteFile;
+static pfnWldpGetApplicationSettingBoolean WldpGetApplicationSettingBoolean;
+static pfnWldpQuerySecurityPolicy WldpQuerySecurityPolicy;
+static PCWSTR NODEJS = L"Node.js";
+static PCWSTR ENFORCE_CODE_INTEGRITY_SETTING_NAME = L"EnforceCodeIntegrity";
+static PCWSTR DISABLE_INTERPRETIVE_MODE_SETTING_NAME =
+  L"DisableInteractiveMode";
+
+void InitWldp(Environment* env) {
+  if (isWldpInitialized) {
+    return;
+  }
+
+  HMODULE wldp_module = LoadLibraryExA(
+      "wldp.dll",
```

Comment:
Is it a default dll on Windows? Can't it be subject of hijack?

I remember that we faced an vulnerability in the past where Node.js was reading the openssl.conf from working_dir instead of absolute openssl path leading to some vulnerabilities. Could you double check if that's the case?

Reply:
This is a default DLL, the OS grabs a handle on this DLL early in the boot process to prevent tampering. 

We can also assume that if this DLL doesn't exist on a system, it is too old to support code integrity features.